http.js

```diff
@@ -393,11 +392,11 @@ exports.read = function (request, qualifier) {
             if (Buffer.isBuffer(data)) {
                 contentType = response.headers['content-type'] || '',
                 match = contentType.match(/charset=([a-z0-9-]+)/i),
-                charset = match ? match[1] : 'utf8';
+                charset = match && iconv.encodingExists(match[1]) ? match[1] : 'utf8';
```

Comment:
Here is the exception the `encodingExists` check prevents:
https://github.com/ashtuchkin/iconv-lite/blob/master/lib/index.js#L102

At any rate, we already have to guess the encoding in case the `charset` parameter is missing. The parameter is explicitly optional in the standard, which recommends falling back to US-ASCII ([RFC 2045](https://www.ietf.org/rfc/rfc2045.txt), section 5.2.). So a guess of some sort is inevitable.

HTTP 1.1 sets the default for user agents to ISO-8859-1 ([RFC 2616](https://www.ietf.org/rfc/rfc2616.txt), section 3.4.1; in practice Windows-1252 is used instead to support the €-sign). Most modern applications seem to assume UTF-8 in the absence of any other information, unless they employ complex detection heuristics.

Since all of the latter ones are byte-compatible supersets of US-ASCII, going with UTF-8 is both standards-conform and future-proof. Arguably it's also the least surprising thing in terms of API behavior (for a convenience method like `read`).